doc/contributing.rst

```diff
@@ -0,0 +1,809 @@
+
+
+.. code-block:: shell
+
+   ((xarray) bash-3.2$ pytest  test_cool_feature.py  -v -k int8
+   =========================== test session starts ===========================
+   platform darwin -- Python 3.6.2, pytest-3.2.1, py-1.4.31, pluggy-0.4.0
+   collected 11 items
+
+   test_cool_feature.py::test_dtypes[int8] PASSED
+   test_cool_feature.py::test_series[int8] PASSED
+
+
+Running the test suite
+----------------------
+
+The tests can then be run directly inside your Git clone (without having to
+install *xarray*) by typing::
+
+    pytest xarray
+
+The tests suite is exhaustive and takes a few minutes.  Often it is
+worth running only a subset of tests first around your changes before running the
+entire suite.
+
+The easiest way to do this is with::
+
+    pytest xarray/path/to/test.py -k regex_matching_test_name
+
+Or with one of the following constructs::
+
+    pytest xarray/tests/[test-module].py
+    pytest xarray/tests/[test-module].py::[TestClass]
+    pytest xarray/tests/[test-module].py::[TestClass]::[test_method]
+
+Using `pytest-xdist <https://pypi.python.org/pypi/pytest-xdist>`_, one can
+speed up local testing on multicore machines. To use this feature, you will
+need to install `pytest-xdist` via::
+
+    pip install pytest-xdist
+
+
+Then, run pytest with the optional -n argument:
+
+    pytest xarray -n 4
+
+This can significantly reduce the time it takes to locally run tests before
+submitting a pull request.
+
+For more, see the `pytest <http://doc.pytest.org/en/latest/>`_ documentation.
+
+Running the performance test suite
+----------------------------------
+
+Performance matters and it is worth considering whether your code has introduced
+performance regressions.  *xarray* is in the process of migrating to
+`asv benchmarks <https://github.com/spacetelescope/asv>`__
+to enable easy monitoring of the performance of critical *xarray* operations.
+These benchmarks are all found in the ``xarray/asv_bench`` directory.  asv
+supports both python2 and python3.
+
+To use all features of asv, you will need either ``conda`` or
+``virtualenv``. For more details please check the `asv installation
+webpage <https://asv.readthedocs.io/en/latest/installing.html>`_.
+
+To install asv::
+
+    pip install git+https://github.com/spacetelescope/asv
+
+If you need to run a benchmark, change your directory to ``asv_bench/`` and run::
+
+    asv continuous -f 1.1 upstream/master HEAD
+
+You can replace ``HEAD`` with the name of the branch you are working on,
+and report benchmarks that changed by more than 10%.
+The command uses ``conda`` by default for creating the benchmark
+environments. If you want to use virtualenv instead, write::
+
+    asv continuous -f 1.1 -E virtualenv upstream/master HEAD
+
+The ``-E virtualenv`` option should be added to all ``asv`` commands
+that run benchmarks. The default value is defined in ``asv.conf.json``.
+
+Running the full test suite can take up to one hour and use up to 3GB of RAM.
```

Comment:
"test suite" -> "benchmark suite"

Can it really take a full hour for xarray? I suspect it's currently much faster, just a few minutes in my experience